datastore/providers/azuresearch_datastore.py

```diff
@@ -0,0 +1,259 @@
+        if delete_all or filter is not None:
+            deleted = set()
+            while True:
+                search_result = await self.client.search(None, filter=filter, top=MAX_DELETE_BATCH_SIZE, include_total_count=True, select=FIELDS_ID)
+                if await search_result.get_count() == 0:
+                    break
+                documents = [{ FIELDS_ID: d[FIELDS_ID] } async for d in search_result if d[FIELDS_ID] not in deleted]
+                if len(documents) > 0:
+                    print(f"Deleting {len(documents)} chunks " + ("using a filter" if filter is not None else "using delete_all"))
+                    del_result = await self.client.delete_documents(documents=documents)
+                    if not all([rr.succeeded for rr in del_result]):
+                        raise Exception("Failed to delete documents")
+                    deleted.update([d[FIELDS_ID] for d in documents])
+                else:
+                    # All repeats, delay a bit to let the index refresh and try again
+                    time.sleep(0.25)
+        
+        if ids is not None and len(ids) > 0:
+            for id in ids:
+                print(f"Deleting chunks for document id {id}")
+                await self.delete(filter=DocumentMetadataFilter(document_id=id))
+
+        return True
+
+    async def _query(self, queries: List[QueryWithEmbedding]) -> List[QueryResult]:
+        """
+        Takes in a list of queries with embeddings and filters and returns a list of query results with matching document chunks and scores.
+        """
+        return await asyncio.gather(*(self._single_query(query) for query in queries))
+    
+    async def _single_query(self, query: QueryWithEmbedding) -> QueryResult:
+        """
+        Takes in a single query and filters and returns a query result with matching document chunks and scores.
+        """
+        filter = self._translate_filter(query.filter) if query.filter is not None else None
+        try:
+            k = query.top_k if filter is None else query.top_k * 2
+            q = query.query if not AZURESEARCH_DISABLE_HYBRID else None
+            if AZURESEARCH_SEMANTIC_CONFIG != None and not AZURESEARCH_DISABLE_HYBRID:
+                print(f"Querying with query: {query.query}, top: {query.top_k}, filter: {filter}, semantic config: {AZURESEARCH_SEMANTIC_CONFIG}")
+                r = await self.client.search(
+                        q, 
+                        filter=filter, 
+                        top=query.top_k, 
+                        vector=Vector(value=query.embedding, k=k, fields=FIELDS_EMBEDDING),
+                        query_type=QueryType.SEMANTIC,
+                        query_language=AZURESEARCH_LANGUAGE,
+                        semantic_configuration_name=AZURESEARCH_SEMANTIC_CONFIG)
+            else:
+                print(f"Querying with query: {query.query}, top: {query.top_k}, filter: {filter}")
```

Comment:
seems like this will log pii and should be removed

Reply:
Good point!

Reply:
Ended up removing this log line entirely. There isn't much telemetry to show in the non-error case outside of the query, no point on adding more noise to logs for the success case.